### ISSUE TITLE
Angular 12: Property 'detach' does not exist on type 'LocalTrack' and  Property 'detach' does not exist on type 'LocalDataTrack'.

### ISSUE THREAD

**Issue by jrayga** [twilio/twilio-video.js#1566]
- [✔] I have verified that the issue occurs with the latest twilio-video.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md).
 - [✔] I reviewed the [Common Issues](https://github.com/twilio/twilio-video.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [✔] I verified that the [Quickstart](https://github.com/twilio/video-quickstart-js) application works in my environment.
 - [✔] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

**Code to reproduce the issue:**

```js
this.twilioRoomSetting.on("disconnected", (room) => {
      console.log("🚀 ~ TwilioVideoCallComponent ~ localRoom", room)
      // Detach the local media elements and stop the track
      room.localParticipant.tracks.forEach((publication) => {
          console.log("🚀 ~ TwilioVideoCallComponent ~ publication", publication)
          publication.track.stop();
          const attachedElements = publication.track.detach();
          attachedElements.forEach(element => element.remove());
          console.log("🚀 ~ TwilioVideoCallComponent ~ attachedElements", attachedElements)
      });
});
```

**Expected behavior:**

 Detach the local media elements and stop the track

**Actual behavior:**

Cannot  Detach the local media elements and stop the track showing errors on compilation and it does show that Property 'detach' does not exist on type 'LocalTrack' and  Property 'detach' does not exist on type 'LocalDataTrack'.

**Software versions:**

 -  Browser(s): MS Edge, Chrome
 -  Operating System: Windows 10
 -  twilio-video.js: 2.16.0
 -  Third-party libraries (e.g., Angular, React, etc.): Angular 12

**Comment by makarandp0** [comment]
Hello @jrayga,

This is expected, because the  `detach` method is applicable only for audio and video tracks and not data tracks.
publication.track has type of [LocalTrack](https://media.twiliocdn.com/sdk/js/video/releases/2.16.0/docs/global.html#LocalTrack`) which can be `LocalDataTrack`
 
You can ensure that you are detaching only audio and video tracks with something like:
```
  room.localParticipant.tracks.forEach((publication) => {
    console.log("🚀 ~ TwilioVideoCallComponent ~ publication", publication)
    if (publication.kind !== 'data' && publication.track ) {
      const track = publication.track as LocalAudioTrack | LocalVideoTrack;
      track.stop();
      const attachedElements = track.detach();
      attachedElements.forEach(element => element.remove());
      console.log("🚀 ~ TwilioVideoCallComponent ~ attachedElements", attachedElements)
    }
  });
```

**Comment by jrayga** [comment]
> LocalAudioTrack | LocalVideoTrack;

Thanks for this one!
